### DISCUSSION TITLE
Setup Inertia JS with Typescript in vue 3 with vite

### DISCUSSION THREAD

**Discussion by JaavierR** [inertiajs/inertia#1383] [Help (Vue)]
Hi everyone,

I have been wanting to configure typescript with inertia for a while now, and now with the release of version 1.0.0 I wanted to test it but I keep running into the same problems. 

The application runs and works without problems, but TS gives me errors that the $page property does not exist on type... for example, maybe it is more with the configuration of TS than inertia, but I tried to look for examples of how to configure everything with vite and I could not find anything, and the documentation does not mention a recommended configuration for TS.

![image](https://user-images.githubusercontent.com/4650238/212500140-b9af1502-d308-41e7-8475-dfafa5cdf47c.png)

If someone could give me some guidance here, or refer me to an example project, or a post in a blog I would greatly appreciate it.

Also thanks to the team for this new release and their amazing work.

**Comment by francoism90** [comment]
@JaavierR You need to create your own types `ineria.d.ts`:

```ts
import type { Page } from '@inertiajs/core'

declare module '@inertiajs/core' {
  interface PageProps {
    auth: {
      logout: {
        route: string
      }
    }
    flash: {
      message: string
    }
  }
}

declare module '@inertiajs/vue3' {
  export declare function usePage<T>(): Page<T>
}
```
